common/src/app/common/types/color.cljc

```diff
@@ -213,11 +236,87 @@
+                             (fill->shape-color fill)
+                             set-fill-color
+                             attach-fill-color
+                             detach-fill-color))
+
+        process-stroke (fn [shape [position stroke]]
+                         (func shape
+                               position
+                               (stroke->shape-color stroke)
+                               set-stroke-color
+                               attach-stroke-color
+                               detach-stroke-color))
+
+        process-shadow (fn [shape [position shadow]]
+                         (func shape
+                               position
+                               (shadow->shape-color shadow)
+                               set-shadow-color
+                               attach-shadow-color
+                               detach-shadow-color))
+
+        process-grid (fn [shape [position grid]]
+                       (func shape
+                             position
+                             (grid->shape-color grid)
+                             set-grid-color
+                             attach-grid-color
+                             detach-grid-color))
+
+        process-text-node (fn [node]
+                            (as-> node $
+                              (reduce process-fill $ (d/enumerate (:fills $)))
+                              (reduce process-stroke $ (d/enumerate (:strokes $)))))
+
+        process-text (fn [shape]
+                       (let [content     (:content shape)
+                             new-content (txt/transform-nodes process-text-node content)]
+                         (if (not= content new-content)
+                           (assoc shape :content new-content)
+                           shape)))]
+
+    (as-> shape $
+      (reduce process-fill $ (d/enumerate (:fills $)))
+      (reduce process-stroke $ (d/enumerate (:strokes $)))
+      (reduce process-shadow $ (d/enumerate (:shadow $)))
+      (reduce process-grid $ (d/enumerate (:grids $)))
+      (process-text $))))
+
+(defn remap-colors
+  "Change the shape so that any use of the given color now points to
+  the given library."
+  [shape library-id color]
+  (let [remap-color (fn [shape position shape-color _ attach-fn _]
+                      (if (= (:ref-id shape-color) (:id color))
+                        (attach-fn shape
+                                   position
+                                   (:id color)
+                                   library-id)
+                        shape))]
+
+    (process-shape-colors shape remap-color)))
+
 (defn sync-shape-colors
   "Look for usage of any color of the given library inside the shape,
   and, in this case, copy the library color into the shape."
   [shape library-id library-colors]
-  (let [sync-color (fn [shape position shape-color set-fn detach-fn]
+  (let [sync-color (fn [shape position shape-color set-fn _ detach-fn]
```

Comment:
The same here, if let only has function definitions, `letfn` is preferible

Reply:
Don't know much what is the advantage. For me it seems more or less the same.

Reply:
Changed anyway.